### PR TITLE
Add docs for new case macros

### PR DESCRIPTION
**Pull request by artememelin**
### Pull request checklist

- [ ] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [x] Did you add new tests?
- [x] Did you add reviewers?

### Added

- Macro `@serde_pascal_case` to transform field names from PascalCase to snake_case for deserialization ([#26](../../pull/26)).
- Macro `@serde_camel_case` to transform field names from camelCase to snake_case for deserialization ([#26](../../pull/26)).
- Macro `@serde_kebab_case` to transform field names from kebab-case to snake_case for deserialization ([#26](../../pull/26)).
- `Serde.ser_name` to override the default field name serialization ([#26](../../pull/26)).
- `Serde.ser_value` to override the default value serialization ([#26](../../pull/26)).
- `Serde.ser_type` to override the default type serialization ([#26](../../pull/26)).
- `Serde.ser_ignore_field` to determine if a field should be ignored during serialization ([#26](../../pull/26)).

### Changed

- Renamed macro `@ser_json_name` to `@ser_name` for consistency with other serialization macros ([#26](../../pull/26)).
- Renamed `ignore_null` to `ser_ignore_null` to align with serialization function naming conventions ([#26](../../pull/26)).
- Renamed `ignore_field` to `ser_ignore_field` for clarity in serialization customization ([#26](../../pull/26)).

### CODE REVIEW

File: src/Utl/Macros.jl [hunk 1]
```diff
@@ -169,6 +169,28 @@ function to_pascal_case(field::String)::String
     return join(titlecase(w) for w in split(field, "_"))
 end
 
+"""
+    @serde_pascal_case T
+
+Marks all fields of type `T` with custom names corresponding to the Pascal case.
```
Comment (dmitrii-doronin) [inline src/Utl/Macros.jl:175]:
```suggestion
Marks all fields of type `T` with custom names corresponding to Pascal case.
```

File: src/Utl/Macros.jl [hunk 2]
```diff
@@ -169,6 +169,28 @@ function to_pascal_case(field::String)::String
...
+
+!!! note
+    Initial field names must match the Snake case.
```
Comment (dmitrii-doronin) [inline src/Utl/Macros.jl:178]:
```suggestion
    Field names type `T` must match Snake case.
```

File: src/Utl/Macros.jl [hunk 3]
```diff
@@ -177,6 +199,28 @@ function to_camel_case(field::String)::String
     return split(field, "_")[1] * join(titlecase(w) for w in split(field, "_")[2:end])
 end
 
+"""
+   @serde_camel_case T
+
+Marks all fields of type `T` with custom names corresponding to the Camel case.
```
Comment (dmitrii-doronin) [inline src/Utl/Macros.jl:205]:
```suggestion
Marks all fields of type `T` with custom names corresponding to Camel case.
```

File: src/Utl/Macros.jl [hunk 4]
```diff
@@ -177,6 +199,28 @@ function to_camel_case(field::String)::String
...
+
+!!! note
+    Initial field names must match the Snake case.
```
Comment (dmitrii-doronin) [inline src/Utl/Macros.jl:208]:
```suggestion
    Field names of type `T` must match Snake case.
```

File: src/Utl/Macros.jl [hunk 5]
```diff
@@ -185,6 +229,28 @@ function to_kebab_case(field::String)::String
     return join([lowercase(w) for w in split(field, "_")], "-")
 end
 
+"""
+    @serde_kebab_case T
+
+Marks all fields of type `T` with custom names corresponding to the Kebab case.
+
+!!! note
+    Initial field names must match the Snake case.
```
Comment (dmitrii-doronin) [inline src/Utl/Macros.jl:238]:
```suggestion
    Field names of type `T` must match Snake case.
```

File: src/Utl/Macros.jl [hunk 6]
```diff
@@ -185,6 +229,28 @@ function to_kebab_case(field::String)::String
     return join([lowercase(w) for w in split(field, "_")], "-")
 end
 
+"""
+    @serde_kebab_case T
+
+Marks all fields of type `T` with custom names corresponding to the Kebab case.
```
Comment (dmitrii-doronin) [inline src/Utl/Macros.jl:235]:
```suggestion
Marks all fields of type `T` with custom names corresponding to Kebab case.
```

**Comment by artememelin** [comment]
DOCS

                                                                https://bhftbootcamp.github.io/Serde.jl/previews/PR28/

DOCS

**Pull request merged**